[antd-migration] migrate from plain css to antd

RawandDev opened the pull request:
Will be working on it later. If you want to work on one of them, comment on this pr.
- [x] content.
- [x] KurdCard.
- [x] Dropdown.

mhmd-azeez commented:
This PR will probably create a merge conflict with #10. Maybe you should wait until we decide on that because I don't want you to throw away all your code. cc @hezhamen

RawandDev commented:
Alright, I will wait.

hezhamen commented:
I agree, I was gonna say we wait for the typescript to be merged.

AramRafeq commented:
@RawandDev  wait until we merge some breaking changes

rojvv commented:
@RawandDev now you can sync your fork resolving the conflicts.

rojvv commented:
@RawandDev you can see the error by running `yarn build`.

RawandDev commented:
@roj1512 I think this is a typescript error and I don't know much about it. I can't use `align="center"`

![image](https://user-images.githubusercontent.com/82733587/147417115-f3e1ccee-7c87-4cb1-bab3-1deab7b24ec0.png)

RawandDev commented:
Feel free to add any suggestions.

rojvv commented:
@RawandDev you fixed errors with TS, that's cool.

rojvv commented on pages/index.tsx:
```diff
@@ -66,45 +69,83 @@ export default function Home() {
+          }}
+        >
+          <Title>Awesome Kurds</Title>
+          <Text type="secondary">
```
This text is appearing too small, in my opinion.

RawandDev replied:
I agree, but if you want to make it bigger I guess you need to use vanilla css. I wanted not to use `style={{...}}` and `className={styles.STYLE_NAME}` as much as I could.

rojvv replied:
Maybe use another variant of `Text`? There's also no issue with `style={{...}}` imho.

RawandDev replied:
There is not any variant that supports bigger text. I think for this issue if you use `style=...` and make the font bigger, then you need to make it smaller on small screens and you can achieve this in `css` which then you need to have `className={styles....}` and make the code messier. By far this is what I have thought of, feel free to commit to this pr tho.

rojvv commented on pages/index.tsx:
```diff
@@ -66,45 +69,83 @@ export default function Home() {
+        >
+          <Title>Awesome Kurds</Title>
+          <Text type="secondary">
+            Meet {awesomeKurds.kurds.length} amazing Kurds.
```
```suggestion
            Meet {awesomeKurds.kurds.length} awesome Kurds.
```

rojvv commented:
@RawandDev you should also resolve the current conflicts.

RawandDev commented:
> you fixed errors with TS, that's cool.

Actually, I did a bit of cheating here, I used `style={{...}}` instead of `align="center`. I prefer setting `align` rather than inline styling. It would be nice if you could check it and see if you can fix it with using `align`. It is in the `Footer`.

rojvv commented:
Sure. Can I get rights to push to your fork?

RawandDev commented:
> Sure. Can I get rights to push to your fork?

Sure.

rojvv commented:
Thanks, @hezhamen! <3

RawandDev commented:
@hezhamen Could you link each pr to its issue? Closing the pr will also close the issue in this way.

rojvv commented:
@RawandDev the `<Footer>` doesn't have an `align=` argument itself. Hence, your current way is not wrong. But you can still put the footer text inside a `<p>` extender or something, and then align that to center instead of its parent.

RawandDev commented:
> @RawandDev the `<Footer>` doesn't have an `align=` argument itself. Hence, your current way is not wrong. But you can still put the footer text inside a `<p>` extender or something, and then align that to center instead of its parent.

Yeah, I think it doesn't have `algin` argument but it is working on localhost. I think the current one is good but I have used some inline styles, and not sure whether to go with `style={{..}}` or `className={styles.NAME}`.

rojvv commented:
@RawandDev as I said, it's better to change it to something like this: `<Footer><Text /* make this align center */>....</Text></Footer>`.

rojvv commented:
I haven't used antd, so you know better about its components.

RawandDev commented:
> @RawandDev as I said, it's better to change it to something like this: `<Footer><Text /* make this align center */>....</Text></Footer>`.

I'm also not familiar with antd but as I did a research you can't also use `align` with `Typograph`s.
```
<Footer><Text style={{ textAlign: "center" }}>....</Text></Footer>
```
You will end up doing pretty much the same with more codes.

rojvv commented:
That's still much better than doing the alignCenter on the Footer.

RawandDev commented:
Not sure if this pr is ignored. If so, please close it without merging 👀

rojvv commented:
@RawandDev no, it is not. But a repository can't stop being updated just because of PRs. Conflicts are easy, believe me.

AramRafeq commented:
Guys there is conflict in one file resolve that and merge it this PR will improve the codebase quality

rojvv commented:
@AramRafeq are you admin of this repo?

rojvv commented:
If you're admin, you can resolve the conflicts yourself and then merge it.

RawandDev commented:
> @RawandDev no, it is not. But a repository can't stop being updated just because of PRs. Conflicts are easy, believe me.

I didn’t say that the repo shouldn’t be updated.
I can resolve conflicts. I know it is not hard but none of the owners seemed to be commenting on this pr recently.

RawandDev commented:
I will resolve the conflicts when I’m on the pc.

AramRafeq commented:
> @AramRafeq are you admin of this repo?

Nope I'm a collaborator @hezhamen  please take the lead

rojvv commented:
@AramRafeq @hezhamen this can get merged now.

RawandDev commented:
I removed some stuff. I saw @hezhamen recently added `.dealer` in css but I couldn't find what was the purpose of it. I removed it and some other unused codes.

hezhamen commented:
.dealer handles our grid for cards

rojvv commented:
@hezhamen can you fix the broken changes yourself?

rojvv commented:
Or maybe I try.

hezhamen commented:
@roj1512 I have made you a collaborator on the repo. You can make the changes yourself. I need to work on some styling issues we currently have.

RawandDev commented:
> .dealer handles our grid for cards

I haven't seen it being used. I also used antd layout so you probably won't need it. Check the deployed version and see if there is any bug.

hezhamen commented:
@RawandDev the cards are too big atm, can you make 5 cards appear per row?

hezhamen commented:
This pr will get merged once someone increases cardPerRow to 5.

RawandDev commented:
@hezhamen I don't think this is achievable in antd. Antd by default doesn't support 5 column grids as far as I know.

rojvv commented:
@hezhamen so you should now decide to agree with no 5 per row or not use antd.

hezhamen commented:
let me check documentation again, and if antd doesn't support it... we'll go with Vanilla CSS for our grid system

rojvv commented:
They don't. Here's a very good confirmation: https://github.com/ant-design/ant-design/issues/18104.

hezhamen commented:
you're right, I'll set up vanilla CSS for it

RawandDev commented:
@roj1512 did you revert the changes then merge the pr? If so, why did you merge this?

rojvv commented:
@RawandDev I just re-added the original styles, and then modified the cards part of the main page. Tell me if something's wrong.

RawandDev commented:
@roj1512 I'm confused xD. Which part exactly did you change? It seems like it doesn't even use antd for layout.

There are only two `Layout` components that is used in `index.tsx` and it doesn't affect anything. Other than that, I think everything is just as it was before.

issue with responsive.
https://gyazo.com/c4c4fdf37aae1c8428886eb74ff41a68

rojvv commented:
@RawandDev can you please send a PR to fix that maintaining the 5 cell per row?
